src/client/WebSocketClient.ts

```diff
@@ -494,7 +499,26 @@ export class WebSocketClient extends EventEmitter {
     return this.socket;
   }
 
-  disconnect(reason: string = 'Unknown reason', forgetSubscriptions: boolean = false): void {
+  disconnect(options?: DisconnectOptions | string): void {
+    let reason = 'Unknown reason';
+    let forgetSubscriptions = false;
+
+    /*
+     * To prevent breaking the legacy API, we allow options to be a string to represent
+     * the reason for the disconnect.
+     */
+    if (typeof options === 'string') {
+      reason = options;
+    } else if (options) {
+      if (undefined !== options.forgetSubscriptions) {
+        forgetSubscriptions = options.forgetSubscriptions;
+      }
```

Comment:
I wish I could change this to just this...

```TypeScript
forgetSubscriptions = options?.forgetSubscriptions ?? false;
```

However, I can't figure out how to do coverage on that.